lib/logstash/outputs/elasticsearch/common.rb

```diff
@@ -91,10 +76,21 @@ def retrying_submit(actions)
                        :backtrace => e.backtrace)
         end
 
-        sleep @retry_max_interval if submit_actions && submit_actions.length > 0
+        # Everything was a success!
+        break if !submit_actions || submit_actions.empty?
+
+        # If we're retrying the action sleep for the recommended interval
+        # Double the interval for the next time through to achieve exponential backoff
+        Stud.stoppable_sleep(sleep_interval) { @stopping.true? }
+        sleep_interval = next_sleep_interval(sleep_interval)
       end
     end
 
+    def next_sleep_interval(current_interval)
+      doubled = current_interval * 2
+      doubled > @retry_max_interval ? @retry_max_interval : doubled
+    end
+
     def submit(actions)
       es_actions = actions.map { |a, doc, event| [a, doc, event.to_hash]}
```

Comment:
I don't see a reason to precompute the actions hash in submit just to avoid the retry mechanism in safe_bulk. I'd rather not have an inlined begin/rescue block in safe_bulk